doc/specs/#1620 - Installer selector on multiple matches.md

```diff
@@ -0,0 +1,93 @@
+---
+author: Darshan Rander @SirusCodes
+created on: 2022-10-29
+last updated: 2022-10-29
+issue id: 1620
+---
+
+# Spec Title
+
+For [#1620](https://github.com/microsoft/winget-cli/issues/1620)"
+
+## Abstract
+
+The winget.exe client simply lists all the matches and if the user wants to install the app they need to re-run the command with Id.
+
+The UX can be improved here by giving users an option to select the package instead of asking them to re-run the command with Id.
+
+## Inspiration
+
+It is not a good experience to copy Id and then run the command again.
+
+## Solution Design
+
+If we find multiple packages then the user should get a selector.
+
+winget should ask for user input, for example -
+
+```pwsh
+> winget install Powershell
+Multiple packages found matching.
+No. Name       Id                   Source
+-------------------------------------------
+ 1  PowerShell 9MZ1SNWT0N5D         msstore
+ 2  PowerShell Microsoft.PowerShell winget
+
+Please select the packages (eg. 2): <user-input>
+
+Installing <user-input>...
+```
+
+> `<user-input>` - is the input by user
+
+After the user will give the input the installation should start.
+
+## UI/UX Design
+
+The idea is explained in [Solution Design](#solution-design).
+
+## Capabilities
+
+### Accessibility
+
+It should not have any direct impact on screen readers.
+
+### Security
+
+It should not have any security issues.
+
+### Reliability
+
+This will help the users to have a better experience and would decrease the chance of human error while copy-pasting the Id.
+
+### Compatibility
+
+It will not break anything.
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
```

Comment:
There is potential confusion when using the `--silent` argument. As a user, I would expect that `--silent` would not lead to an interactive menu. Having reviewed the spec and being familiar with how Winget works, I understand that the selector is not part of the package install, but this could be a sticking point for many new users and should be preempted.

Reply:
Yeah, we should not show selector when `--silent` argument is passed.

Reply:
> Yeah, we should not show selector when `--silent` argument is passed.

I think more opinions would be needed here. IMO `--silent` shouldn't control how winget behaves as a CLI, that's what the `--disable-interactivity` prompt is for. My point was more that some mention should be made around this, or it should be documented that `--silent` doesn't suppress this and that `--disable-interactivity` would. @yao-msft and @ryfu-msft would be the experts here though